internal/fs/buffer/inmemory_write_buffer_test.go

```diff
@@ -41,24 +56,371 @@ func (t *MemoryBufferTest) SetUp(ti *TestInfo) {
 
 func (t *MemoryBufferTest) TearDown() {}
 
+// //////////////////////////////////////////////////////////////////////
+// Helpers
+// //////////////////////////////////////////////////////////////////////
+
+func (t *MemoryBufferTest) generateRandomData(size int64) []byte {
+	r := rand.New(rand.NewSource(time.Now().UnixNano()))
+	data := make([]byte, size)
+	_, err := r.Read(data)
+	AssertEq(nil, err)
+	return data
+}
+
+func (t *MemoryBufferTest) validateInMemoryBuffer(buffer inMemoryBuffer,
+	capacity, length, startOffset, endOffset int64) {
+	AssertEq(capacity, cap(buffer.buffer))
+	AssertEq(length, len(buffer.buffer))
+	AssertEq(startOffset, buffer.offset.start)
+	AssertEq(endOffset, buffer.offset.end)
+}
+
+func (t *MemoryBufferTest) createAndValidateInMemoryBuffer(sizeInMB int64) {
+	var err error
+	t.mb, err = CreateInMemoryWriteBuffer(uint(sizeInMB))
+
+	AssertEq(nil, err)
+	AssertEq(nil, t.mb.current.buffer)
+	AssertEq(nil, t.mb.flushed.buffer)
+	AssertEq(sizeInMB*MiB, t.mb.bufferSize)
+	AssertEq(0, t.mb.fileSize)
+}
+
 // //////////////////////////////////////////////////////////////////////
 // Tests
 // //////////////////////////////////////////////////////////////////////
 
-func (t *MemoryBufferTest) TestCreateEmptyInMemoryBuffer() {
-	t.mb = CreateInMemoryWriteBuffer()
+func (t *MemoryBufferTest) TestCreateEmptyInMemoryBufferWithValidBufferSize() {
+	t.createAndValidateInMemoryBuffer(bufferSizeInMB)
+}
+
+func (t *MemoryBufferTest) TestCreateEmptyInMemoryBufferWith0BufferSize() {
+	var err error
+	var sizeInMB int64 = 0
+	t.mb, err = CreateInMemoryWriteBuffer(uint(sizeInMB))
+
+	AssertNe(nil, err)
+	AssertTrue(strings.Contains(err.Error(), ZeroSizeBufferError))
+	AssertEq(nil, t.mb)
+}
+
+func (t *MemoryBufferTest) TestWritingToUninitializedBuffer() {
+	// Write to buffer
+	err := t.mb.WriteAt([]byte(smallContent1), 0)
+
+	AssertNe(nil, err)
+	AssertTrue(strings.Contains(err.Error(), ZeroSizeBufferError))
+}
+
+func (t *MemoryBufferTest) TestEnsureCurrentBuffer() {
+	t.createAndValidateInMemoryBuffer(bufferSizeInMB)
+
+	err := t.mb.ensureCurrentBuffer()
+
+	AssertEq(nil, err)
+	AssertEq(bufferSizeInMB*MiB, t.mb.bufferSize)
+	t.validateInMemoryBuffer(t.mb.current, bufferSizeInMB*MiB, 0, 0, bufferSizeInMB*MiB)
```

Comment:
`bufferSizeInMB*MiB` appears too many times in this file. Could you just rename this to `bufferSize` or `bufferSizeInBytes` ?